.github/workflows/ci.yaml

```diff
@@ -0,0 +1,20 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        strategy:
+            matrix:
+                node-version: [18.x, 20.x]
+
+        steps:
+            - uses: actions/checkout@v4
+            - name: Use Node.js ${{ matrix.node-version }}
+              uses: actions/setup-node@v3
+              with:
+                  node-version: ${{ matrix.node-version }}
+            - run: yarn install --frozen-lockfile
+            - run: yarn test
```

Comment:
While we're setting up tests. We can also integrate ESLint here which should check for lint errors and also enforce styling in case pre-commit hook is skipped. Not part of this PR, can be done separately